### PR TITLE
A couple of WCS DistortionLookupTable fixes

### LINKED ISSUE #13456: Possible off-by-one error in WCS DistortionLookupTable
### Description
I think I've found an error in the `DistortionLookupTable` class in `astropy.wcs`. That class implements "distortion paper-style" lookup tables for distortions to be applied to pixel coordinates before they are converted to world coordinates. The distortion lookup table standard uses a set of `crpix`/`crval`/`cdelt` values for relating distortion table pixel coordinates to image array pixel coordinates (so that there need not be a distortion table pixel for each image array pixel---instead, interpolation can be used with a coarser grid of distortion values).

The values of `crpix` and `crval` are treated as zero-based indices, i.e. the the first row and column are number zero. However, when FITS files use `crpix` in the main WCS for converting image array coordinates to world coordinates, those values are instead one-based indices, i.e. the first row and column are number one. The values stored and accessible in `my_WCS_object.wcs.crpix` are also one-based. I haven't found it stated explicitly anywhere that the FITS-header values of `crpix` and `crval` for the distortion array <-> image array transformation are also one-based, but that seems like a reasonable and consistent assumption. However, the current implementation in `astropy.wcs` treats these values as zero-based indices, both when reading FITS files and when manually creating `DistortionLookupTable` instances. If I'm right in assuming the values should be treated as one-based indices, this is a bug in `DistortionLookupTable`. If I'm wrong, `DistortionLookupTable` is correct as-is, though I might suggest noting this difference in `crpix`/`crval` semantics in the documentation.

To go into some more depth:

The zero-index nature of `DistortionLookupTable` appears to be implied in a few parts of `astropy/wcs/wcs.py` such as [this](https://github.com/astropy/astropy/blob/19cc80471739bcb67b7e8099246b391c355023ee/astropy/wcs/wcs.py#L1001) line, where default values of 0 are used for `crpix` and `crval` when reading the header of the distortion table in a FITS file. Note also that the values read from the header are directly passed to `DistortionLookupTable`, and I couldn't find any part of `DistortionLookupTable` that changes those values, suggesting the indexing base for `DistortionLookupTable`'s arguments is assumed to be the same as what's in the FITS file---though FITS headers usually use one-based indices afaik.

The indexing base comes into play in `astropy/wcs/src/distortion.c` within `image_coord_to_distortion_coord` [(here)](https://github.com/astropy/astropy/blob/19cc80471739bcb67b7e8099246b391c355023ee/astropy/wcs/src/distortion.c#L84) where the comment indicates "The '- 1./stepsize' is here because the input coordinates are 1-based, but this is a C-array underneath". This function converts pixel coordinates in the data array to pixel coordinates in the distortion array, with an expression that translates to
```
dist_arr_coord = (img_arr_coord - crval) / cdelt + crpix - 1 / cdelt
```
or equivalently,
```
dist_arr_coord = (img_arr_coord - 1 - crval) / cdelt + crpix
```
The -1 converts one-based pixel coordinates to zero-based array coordinates, and by applying it before the `crval`/`cdelt`/`crpix` combo, it is assumed that `crpix` and `crval` are zero-based.

To summarize, when loading a FITS file containing distortion lookup tables, the `crpix` and `crval` values from those tables' headers are passed unaltered into `DistortionLookupTable`, and I haven't found any place within `DistortionLookupTable` that alters those values. Those values are used as if they are zero-based indices, but I think it is reasonable to assume the values in the FITS headers are intended to be one-based indices.

If I'm correct that this is an inconsistency, one could subtract 1 from `crpix` and `crval` when reading the distortion table FITS headers in `astropy/wcs/wcs.py`. But when creating WCS objects from scratch (as I do in the example below), the `crpix` values to be given are one-based, so I think it makes sense to treat them as one-based in `DistortionLookupTable` as well, which means `astropy/wcs/src/distortion.c` should be modified to assume one-based values. (And perhaps the default values of 0 in `astropy/wcs/wcs.py` should be changed to 1.)

### Expected behavior
When creating `DistortionLookupTables`, `crpix` and `crval` should be treated as one-based indices, so that values of `1, 1` result in the first row/column of the distortion table applying to the first row/column of the image array, giving an arrangement like this:
```
distortion array values:   | 0 |       |-.3|       |-.6|
distortion array indices:  | 1 |       | 2 |       | 3 |
data array indices:        | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 |
```

### Actual behavior
When creating `DistortionLookupTables`, `crpix` and `crval` are treated as zero-based indices, so that values of `0, 0` result in the first row/column of the distortion table applying to the first row/column of the image array. A value of `1, 1` causes a misalignment in that transformation, dependent on the value of `cdelt`. (`cdelt = 1, 1` hides the issue.)

Values of `1,1` given an arrangement like this:
```
distortion array values:   | 0 |       |-.3|       |-.6|
distortion array indices:  | 0 |       | 1 |       | 2 |
data array indices:                | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 |
```

Values of `0,0` give an arrangement like this:
```
distortion array values:   | 0 |       |-.3|       |-.6|
distortion array indices:  | 0 |       | 1 |       | 2 |
data array indices:        | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 |
```

### Steps to Reproduce

```python
from astropy.wcs import WCS, DistortionLookupTable
import numpy as np

shape = 500, 500

# Create a demo WCS
wcs_orig = WCS(naxis=2)
wcs_orig.wcs.crpix = 250, 250
wcs_orig.wcs.crval = 250, 250
wcs_orig.wcs.cdelt = 1, 1

# Define our distortion table at these x and y points
x = np.arange(0, shape[1], 3)
y = np.arange(0, shape[0], 3)

# Create an x offset that varies linearly with x
x_offset = -0.1 * x
# Stack it to apply the same x offset at all y values
distortions_x = np.vstack([x_offset] * len(y))

# Create a y offset that varies linearly with y
y_offset = -0.2 * y
# Stack it to apply the same y offset at all x values
distortions_y = np.vstack([y_offset] * len(x)).T

# Build the distortion tables
crpix = 1, 1
crval = 1, 1
cdelt = x[1] - x[0], y[1] - y[0]

map_x = DistortionLookupTable(distortions_x.astype(np.float32),
                              crpix,
                              crval,
                              cdelt)
map_y = DistortionLookupTable(distortions_y.astype(np.float32),
                              crpix,
                              crval,
                              cdelt)

wcs_distorted = wcs_orig.deepcopy()
wcs_distorted.cpdis1 = map_x
wcs_distorted.cpdis2 = map_y

# we can manually compute what the distortion offset should be for a location and apply it to
# coordinates that we pass into the undistorted `wcs_orig.pixel_to_world`. These should
# match the coordinates we get from `wcs_distorted.pixel_to_world`.

dx = np.interp(xi, x, x_offset)
dy = np.interp(yi, y, y_offset)

print(f"For xi={xi}, yi={yi}, computed offsets are dx={dx}, dy={dy}")

print(wcs_distorted.pixel_to_world(xi, yi))
print(wcs_orig.pixel_to_world(xi + dx, yi + dy))

# Output:
# [<Quantity 225.79999924>, <Quantity 200.59999847>]
# [<Quantity 226.>, <Quantity 201.>]
# They don't match! But if we re-do the distortion tables with `crpix` and `crval` set to `0, 0`, they do match:

crpix = 0, 0
crval = 0, 0

map_x = DistortionLookupTable(distortions_x.astype(np.float32),
                              crpix,
                              crval,
                              cdelt)
map_y = DistortionLookupTable(distortions_y.astype(np.float32),
                              crpix,
                              crval,
                              cdelt)

wcs_distorted_zero_indexed = wcs_orig.copy()

wcs_distorted_zero_indexed.cpdis1 = map_x
wcs_distorted_zero_indexed.cpdis2 = map_y

print(wcs_distorted_zero_indexed.pixel_to_world(xi, yi))
print(wcs_orig.pixel_to_world(xi + dx, yi + dy))

# Output:
# [<Quantity 226.>, <Quantity 201.>]
# [<Quantity 226.>, <Quantity 201.>]
```

### System Details
```
macOS-12.3.1-arm64-arm-64bit
Python 3.9.13 | packaged by conda-forge | (main, May 27 2022, 17:00:33) 
[Clang 13.0.1 ]
Numpy 1.22.4
pyerfa 2.0.0.1
astropy 5.2.dev271+gfe3486171.d20220714
Scipy 1.8.1
Matplotlib 3.5.2
```

### LINKED ISSUE #15047: wcs.DistortionLookupTable not handled when slicing
### Description

When slicing a `WCS` object that includes distortion lookup tables (i.e. `my_wcs.cpdis1` or `cpdis2` is a `wcs.DistortionLookupTable`), the lookup table is not modified by the slice, so it applies to the wrong location in the data array.

I'm not opposed to trying to make a PR to fix this, but I want to make sure I'm not missing anything first.

### Expected behavior

The `CRVAL` of the distortion lookup table headers should be modified, similar to if not exactly the same as how the `CRPIX` values of the main data WCS are modified, so that the distortions in the lookup table apply to the same parts of the data array.

### How to Reproduce

```python
import numpy as np
from astropy.wcs import WCS, DistortionLookupTable

# Create a demo WCS, where world coordinates are just pixel coords + 100
wcs = WCS(naxis=2)
wcs.wcs.crpix = 101, 101
wcs.wcs.crval = 200, 200
wcs.wcs.cdelt = 1, 1

# Create a distortion lookup table, which pushes pixels past pixel number 100 out by 50 pixels
distortion = np.zeros((200, 200), dtype=np.float32)
distortion[100:] = 50
dlt = DistortionLookupTable(distortion, (1,1), (1,1), (1,1))

# Copy our WCS and add the lookup table for both x and y coordinates
wcs_distorted = wcs.deepcopy()
wcs_distorted.cpdis1 = dlt
wcs_distorted.cpdis2 = dlt

# Demo the effect of the lookup table---coordinates are offset by 50, starting at pixel 100
print("Undistorted", wcs.pixel_to_world(99, 99), wcs.pixel_to_world(100, 100))
print("Distorted", wcs_distorted.pixel_to_world(99, 99), wcs_distorted.pixel_to_world(100, 100))

# Output:
# Undistorted [<Quantity 199.>, <Quantity 199.>] [<Quantity 200.>, <Quantity 200.>]
# Distorted [<Quantity 199.>, <Quantity 199.>] [<Quantity 250.>, <Quantity 250.>]

# If we slice the WCS (cropping off the first 50 rows and columns of the data array), the undistorted wcs still assigns
# the same coordinates to the same image pixels, but the distorted array has changed its output for pixel (50, 50) in
# the cropped array (which is (100, 100) in the uncropped array).
print("Undistorted, sliced", wcs[50:, 50:].pixel_to_world(49, 49), wcs[50:, 50:].pixel_to_world(50, 50))
print("Distorted, sliced", wcs_distorted[50:, 50:].pixel_to_world(49, 49), wcs_distorted[50:, 50:].pixel_to_world(50, 50))

# Output:
# Undistorted, sliced [<Quantity 199.>, <Quantity 199.>] [<Quantity 200.>, <Quantity 200.>]
# Distorted, sliced [<Quantity 199.>, <Quantity 199.>] [<Quantity 200.>, <Quantity 200.>]

# And if we go to (100, 100) in the cropped array, we see the 50-pixel offset from the distortion lookup table---it's not
# accounting for the sliced WCS or cropped data array
print(wcs_distorted[50:, 50:].pixel_to_world(99, 99), wcs_distorted[50:, 50:].pixel_to_world(100, 100))

# Output:
# [<Quantity 249.>, <Quantity 249.>] [<Quantity 300.>, <Quantity 300.>]
```

I think the fix will involve having `WCS.slice` update the `CRVAL` in the `DistortionLookupTable`, sort of like this:
```python
wcs_distorted_sliced_fixed = wcs_distorted[50:, 50:]
wcs_distorted_sliced_fixed.cpdis1.crval -= 50

print("Undistorted, sliced", wcs[50:, 50:].pixel_to_world(49, 49), wcs[50:, 50:].pixel_to_world(50, 50))
print("Distorted, sliced, fixed", wcs_distorted_sliced_fixed.pixel_to_world(49, 49), wcs_distorted_sliced_fixed.pixel_to_world(50, 50))

# Output:
# Undistorted, sliced [<Quantity 199.>, <Quantity 199.>] [<Quantity 200.>, <Quantity 200.>]
# Distorted, sliced, fixed [<Quantity 199.>, <Quantity 199.>] [<Quantity 250.>, <Quantity 250.>]
```

### Versions

Linux-4.18.0-305.12.1.el8_4.x86_64-x86_64-with-glibc2.28
Python 3.10.11 | packaged by conda-forge | (main, May 10 2023, 18:58:44) [GCC 11.3.0]
astropy 5.3
Numpy 1.24.3
pyerfa 2.0.0.3
Scipy 1.10.1
Matplotlib 3.6.3

### PR DESCRIPTION
**Pull request by svank**
This fixes #13456 and fixes #15047

The first commit adds a few tests for `DistortionLookupTable`, which looks like it had very little coverage before this. I've tried to make it easy to be sure that the tested-for behavior is correct. Both tests will fail without the following commits.

* #13456 was a long, stream-of-conciousness bug report (I wrote it, so I can say that). I don't think my "off-by-one" analysis then fully holds water. Looking at things again, I think `DistortionLookupTable`'s CRPIX and CRVAL are treated consistently through the code (as FITS-style one-based indices), but there is a problem with how the code calculates the distortion-table pixel corresponding to an image-pixel location. At the location of the code change, the image pixel coordinate and the `DistortionLookupTable`'s CRPIX and CRVAL are all 1-based indices, and so the conversion to a zero-based index that the comment there mentions should happen at the "end" of the calculation and should not be scaled by CDELT. With this change, one of the new tests passes.

* #15047 noted that `DistortionLookupTables` simply aren't handled when slicing a WCS. This PR implements that handling. I've also added handling for the "det2im" tables, which I'm not familiar with, but as far as I can tell they're just another name for "cpdis" distortion maps, so I've treated them the same way.

### CODE REVIEW

File: astropy/wcs/docstrings.py [hunk 1]
```diff
@@ -1024,7 +1024,7 @@ def RA_DEC_ORDER(indent=0):
 """
 
 DistortionLookupTable = """
-DistortionLookupTable(*table*, *crpix*, *crval*, *cdelt*)
+DistortionLookupTable(table, crpix, crval, cdelt)
```
Comment (pllim) [inline astropy/wcs/docstrings.py:1027]: No need to remove the formatting here?

**Comment by pllim** [comment]
Thanks! Would need a change log too.

For pre-commit, please see https://docs.astropy.org/en/latest/development/workflow/maintainer_workflow.html#pre-commit-bot

I tentatively not going to backport but if WCS maintainer(s) think it is safe to backport, they can change the milestone and slap the backport label on.

File: astropy/wcs/docstrings.py (hunk 1 above)
Comment (svank) [reply astropy/wcs/docstrings.py:1027]: I'm happy to leave it, it just looked very non-standard---it comes through in the [HTML docs](https://docs.astropy.org/en/stable/api/astropy.wcs.DistortionLookupTable.html#astropy.wcs.DistortionLookupTable), where I've never seen asterisks like that before.

Comment (pllim) [reply astropy/wcs/docstrings.py:1027]: Oh, I thought it was to make it *italic* but maybe that isn't the case. OK, let's leave your removal as-is and see what `wcs` maintainers say. Thanks!

**Comment by nden** [comment]
The FITS standard states that image coordinates are 1-based and default values of CRPIX and CRVAL is 0. Does this change your analysis?

**Comment by svank** [comment]
Oh, I didn't know the default values are set in the standard. When I wrote up #13456, those defaults had been giving a zero-based "flavor" for me, so it's good to know they're intended defaults for the one-based values.

Looking at this a little more, in `distortion.c`, `image_coords_to_distortion_coords` is translating from an image-plane coordinate to a distortion array coordinate, and outputting a zero-based index into the array storing the distortion lookup table. Before this PR, the equation rearranges to `dist_arr_coord = (img_arr_coord - 1 - crval) / cdelt + crpix`, which would be correct if `crval` and `crpix` were both zero-based, while `img_arr_coord` were one-based. But I'm pretty sure that at this point, `img_arr_coord` is always one-based, and `crval` and `crpix` are as they were in the FITS header, so also one-based, meaning the modified equation in this PR is correct: `dist_arr_coord = (img_arr_coord - crval) / cdelt + crpix - 1`

File: astropy/wcs/src/distortion.c [hunk 2]
```diff
@@ -79,11 +79,11 @@ image_coord_to_distortion_coord(
   assert(lookup != NULL);
   assert(axis < NAXES);
 
-  /* The "- 1./stepsize" is here because the input coordinates are 1-based,
+  /* The "- 1" is here because the input coordinates are 1-based,
      but this is a C-array underneath */
   result = (
       ((img - lookup->crval[axis]) / lookup->cdelt[axis]) +
-      lookup->crpix[axis]) - 1.0/lookup->cdelt[axis];
+      lookup->crpix[axis]) - 1.0;
```
Comment (nden) [inline astropy/wcs/src/distortion.c:86]: I've been thinking about this change, it's been a while since I looked at this code so it took me a while.
Suppose the lookup table (LT) is two times smaller than the image, i.e. cdelt=2. For simplicity let's take CRPIX=0 and CRVAL=0. Coordinates are 1-based because this is FITS. The first pixel in the image array, img_x=1, should map into the first pixel of the LT, lt_x=1. And because it's a C array this corresponds to index 0. So `img_x/cdelt + correction` should result in 0 (ignoring crpix and crval). `correction` needs to be `-1/cdelt` to get 0. This is because 1 pixel in the image corresponds to 1/cdelt pixels in the LT. 
The first two pixels in the image array map into the first pixel in the LT, and so on. img_x=3 will map into index 1, i.e. lt_x=2.
Similarly, going in the opposite direction, from LT to image, there needs to be a correction to get the C array index and in this case the correction is `-cdelt`, or `-1*cdelt`, because 1 pixel in the LT corresponds to cdelt pixels in the image. For example, if cdelt was 64, the correction going from LT to image should be -64.

I hope this makes sense. The original is correct in my opinion.
To summarize, internally all computations assume 1-based coordinates. To get an index into the array, a correction is applied and it is equal to the fractional pixel in the LT corresponding to a full pixel in the image.

Comment (svank) [reply astropy/wcs/src/distortion.c:86]: Thanks for looking at this! There are two spots where I'm not following along:

If I'm understanding right, it seems like you're applying the conversion from FITS pixels to array indices (the offset of 1) in image space and not LT space, but I think LT space is where it should happen. In this function we have a one-based img_x as input, and we want the corresponding value from the LT array, so a workflow of converting to a 1-based lt_x, and then subtracting 1 to get an array index makes sense to me. Whereas this code (before this PR) translates to `lt_array_index = (img_x - crval) / cdelt + crpix - 1/cdelt` => `lt_array_index = ([img_x - 1] - crval) / cdelt + crpix`, which I interpret as converting img_x to an array index into the image, and then converting to a LT index, but that's done with the 1-based crpix and crval, which doesn't seem right.

CRPIX and CRVAL must be 1-based coordinates, right? That's how `WCS` seems to handle it on the image<->world side. So when we set crpix=crval=0, it's not the case that img_x=1 corresponds to lt_x=1. Instead, shouldn't lt_x=1 correspond to img_x=2 (while lt_x=0.5 is img_x=1 and lt_x=0 is img_x=0)? So then `img_x/cdelt + correction` is `2/2 + correction`, and to get array index 0 corresponding to lt_x=1, we need `correction=-1`. This gives `[(img_x - crval) / cdelt + crpix] - 1`, which to me has the straightforward interpretation of converting the 1-based img_x to a 1-based lt_x with everything in the square brackets, and then converting lt_x to a 0-based array index into the LT array.

Comment (nden) [reply astropy/wcs/src/distortion.c:86]: > it seems like you're applying the conversion from FITS pixels to array indices in image space and not LT space

Yes, I think the offset needs to be applied in `input` space. Essentially the correction needs to be one input pixel expressed in output coordinates. I think words explain this better than numbers. In either direction, the first element of the image array needs to map into the first element of the LT array, and the last element of the image into the last element of the LT. In between we interpolate. 

> So when we set crpix=crval=0, it's not the case that img_x=1 corresponds to lt_x=1. Instead, shouldn't lt_x=1 correspond to img_x=2

I think not, see above. When you design the look up table and determine what the WCS keywords need to be, the starting point is that he first and last elements need to map. This determines the CRVAL and CRPIX.

Comment (nden) [reply astropy/wcs/src/distortion.c:86]: Is there a bigger story behind this? The argument here is which two pixels are used in the interpolation, with a difference of one pixel. Unless the differences in adjacent pixels are large the end differences should be negligible.

Comment (svank) [reply astropy/wcs/src/distortion.c:86]: > Is there a bigger story behind this? The argument here is which two pixels are used in the interpolation, with a difference of one pixel. Unless the differences in adjacent pixels are large the end differences should be negligible.

You're right that this should be a very small difference, but I still think it's something that astropy should get right and not just close enough. I care as part of trying to align PSP/WISPR images with as much sub-pixel precision as possible, to support a starfield estimation and removal process, so I've been focusing on small details. I must have come across this (over a year ago) while testing that I was building and using `DistortionLookupTable`s correctly.

> In either direction, the first element of the image array needs to map into the first element of the LT array, and the last element of the image into the last element of the LT

This in particular has implications for the other half of this PR, handling the LT during wcs slicing. I agree that a well-designed LT will have its first and last pixels map to the first and last image pixels, which then constrains the values of CRPIX/VAL/DELT, but the way I read the Calabretta paper laying this all out, this isn't required, and the first/last LT pixels could map to somewhere beyond the bounds of the image. They call this out on p10 as a way to handle the LT when extracting a sub-image from the main image, by just updating the LT's CRVAL, and only optionally cropping out the LT pixels that don't map anywhere in the subimage and updating CRPIX to match (as I understand those two sentences).

So I really think that the CRPIX, VAL = 0, 0 from your example means that the LT and image are "aligned" at their 0th pixels or -1st array indices (if negative indices didn't wrap around), and then the first actual LT pixel maps to the second image pixel (since cdelt=2). And I think that still implies that `correction` must be `-1`, as in my earlier comment. To have the first image pixel map to the first LT pixel, while CDELT != 1, the choice would be CRPIX, VAL = 1, 1 (or `1+n, 1+n*cdelt`), which will also require `correction=1` to produce the right array indices.

If you still don't agree, I wonder if you could look at the `test_DistortionLookupTable` I added in this PR. (I just reorganized it a bit so it was easier to double-check.) The test passes only with this PR in place---what would you change so it passes without this PR?

Again, thanks for your time on this!

**Comment by mcara** [comment]
My apologies to taking time to understand this issue as I am no black belt in WCS though now I can add Paper IV to my belt. After looking through Paper IV, I agree with @svank that the formula used to computing indices into LT (implemented in `image_coord_to_distortion_coord()`) is incorrect and I am confident that the proposed fix in this PR is the correct one.

**Review by mcara — APPROVED** [review]
Looks good to me. Thanks!

File: docs/changes/wcs/16163.bugfix.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+``DistortionLookupTable`` s (which implement ``cpdis`` and ``det2im`` projection corrections to a WCS) are now correctly aligned to the image plane, and they are handled when a WCS is sliced.
```
Comment (pllim) [inline docs/changes/wcs/16163.bugfix.rst:1]: The orphaned "s" looks weird and I am not 100% sure removing the whitespace would break formatting or not. Is changing to singular ok?
 
```suggestion
``DistortionLookupTable`` (which implements ``cpdis`` and ``det2im`` projection corrections to a WCS) is now correctly aligned to the image plane, and they are handled when a WCS is sliced.
```

Comment (mcara) [reply docs/changes/wcs/16163.bugfix.rst:1]: I think removing white space should be fine. It could also be reformulated as:
```suggestion
``DistortionLookupTable`` objects (which implement ``cpdis`` and ``det2im`` projection corrections to a WCS) are now correctly aligned to the image plane, and they are handled when a WCS is sliced.
```

Comment (mcara) [reply docs/changes/wcs/16163.bugfix.rst:1]: However, I think "are now correctly aligned to the image plane" is a little confusing (especially for me thinking about image _alignment_). I would like to suggest something along these lines (although it feels long and could be shortened with details):
```suggestion
Fixed a bug in ``DistortionLookupTable`` (which implement ``cpdis`` and ``det2im`` projection corrections to a WCS) due to which image coordinates were mapped incorrectly into the lookup table (~1 index offset) resulting in incorrect computation of distortions based on neighboring elements in the table.

Added support for slicing WCS objects containing ``DistortionLookupTable``  which previously were ignored.
```

I also feel last line is not a bug fix but rather an enhancement and probably should be moved to a separate change log entry.

Comment (svank) [reply docs/changes/wcs/16163.bugfix.rst]: I've tried to keep the idea but make it more succinct, thanks!

**Comment by svank** [comment]
@mcara thanks for looking at this!

Just for interest's sake, I was looking at when this `DistortionLookupTable` code was added---it was PR #13, bringing pywcs into astropy. The second comment there says "I don't expect a full review -- I'm sure we'll find other problems over time." And now here we are, 13 years later... :sweat_smile:

File: docs/changes/wcs/16163.bugfix.rst (hunk 3 above)
Comment (pllim) [reply docs/changes/wcs/16163.bugfix.rst]: LGTM now. Thanks!
